elementary/monitor/api/alerts/alerts.py

```diff
@@ -0,0 +1,319 @@
+import json
+from collections import defaultdict
+from datetime import datetime
+from typing import Callable, Dict, List, Tuple
+
+from elementary.clients.api.api import APIClient
+from elementary.clients.dbt.dbt_runner import DbtRunner
+from elementary.config.config import Config
+from elementary.monitor.alerts.alert import Alert
+from elementary.monitor.alerts.alerts import Alerts, AlertsQueryResult
+from elementary.monitor.alerts.malformed import MalformedAlert
+from elementary.monitor.alerts.model import ModelAlert
+from elementary.monitor.alerts.source_freshness import SourceFreshnessAlert
+from elementary.monitor.alerts.test import TestAlert
+from elementary.monitor.api.alerts.normalized_alert import NormalizedAlert
+from elementary.utils.log import get_logger
+from elementary.utils.time import DATETIME_FORMAT, get_now_utc_str
+
+logger = get_logger(__name__)
+
+FALLBACK_TIME = datetime.fromtimestamp(0).strftime(DATETIME_FORMAT)
+
+
+class AlertsAPI(APIClient):
+    def __init__(
+        self, dbt_runner: DbtRunner, config: Config, elementary_database_and_schema: str
+    ):
+        super().__init__(dbt_runner)
+        self.config = config
+        self.elementary_database_and_schema = elementary_database_and_schema
+
+    def get_test_alerts(self, days_back: int, disable_samples: bool = False):
+        pending_test_alerts = self._query_pending_test_alerts(
+            days_back, disable_samples
+        )
+        last_alert_sent_times = self._query_last_test_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_test_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=TestAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_model_alerts(self, days_back: int):
+        pending_model_alerts = self._query_pending_model_alerts(days_back)
+        last_alert_sent_times = self._query_last_model_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_model_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=ModelAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_source_freshness_alerts(self, days_back: int):
+        pending_source_freshness_alerts = self._query_pending_source_freshness_alerts(
+            days_back
+        )
+        last_alert_sent_times = self._query_last_source_freshness_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_source_freshness_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=SourceFreshnessAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_new_alerts(self, days_back: int, disable_samples: bool = False) -> Alerts:
```

Comment:
Get new alerts used to call 3 different methods (one fore each alert type).
It should be like that IMO (more similar to a real API) and it is really good structure for supporting multiple alert types.

I updated the get specific alert type methods to support suppression